### PR TITLE
[SAIC-569] Verify openstack objects of deleted tenant on DST

### PR DESCRIPTION
**Pull request by pavloivanets**
Migrate deleted tenant's openstack objects.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.test_verify_dst_del_tenant_resources
devlab/tests/test_verify_dst_del_tenant_resources.py:110: [W0613(unused-argument), VerifyDstDeletedTenantResources.volume_is_attached_to_vm] Unused argument 'vol_name'
devlab/tests/test_verify_dst_del_tenant_resources.py:126: [E1101(no-member), VerifyDstDeletedTenantResources.cinder_volumes_exist_on_dst] Instance of 'int' has no 'append' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.test_verify_dst_del_tenant_resources
devlab/tests/test_verify_dst_del_tenant_resources.py:108: [W0613(unused-argument), VerifyDstDeletedTenantResources.volume_is_attached_to_vm] Unused argument 'vol_name'
devlab/tests/test_verify_dst_del_tenant_resources.py:124: [E1101(no-member), VerifyDstDeletedTenantResources.cinder_volumes_exist_on_dst] Instance of 'int' has no 'append' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        # according to current implementation,all tenant's vms
+        # should be moved tothe 'admin' account
+        src_vm_names = [vm['name'] for vm in src_vm_list]
+        migrated_vm_list = []
+        for dst_vm in self.dst_vm_list:
+            for src_vm in src_vm_list:
+                if src_vm['name'] == dst_vm.name:
+                    if self.tenants['admin'] == dst_vm.tenant_id:
+                        migrated_vm_list.append(dst_vm.name)
+        result = bool(len(migrated_vm_list) == len(src_vm_list))
```
Comment (roman-verchikov) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: `bool()` conversion is unnecessary, `==` operator returns boolean value

**Comment by roman-verchikov** [comment]
@pavloivanets 
1. Remove all unnecessary `bool()` conversions
2. Rename non-test methods as private (all methods which don't start with `test_` should be started with `_` (underscore))
3. Update commit message and pull request description to be readable

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.test_verify_dst_del_tenant_resources
devlab/tests/test_verify_dst_del_tenant_resources.py:103: [E0602(undefined-variable), VerifyDstDeletedTenantResources.test_tenant_vm_exists_on_dst] Undefined variable 'undeleted_vms'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1522/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            [user for user in config.users if
+             'tenant' in user and user['tenant'] == TEST_TENANT_NAME]
+
+        user_names_tenant = [user['name'] for user in self.tenant_users_list]
+        user_names_with_keypair = [key['user'] for key in config.keypairs]
+        self.tenant_users_with_keyp = \
+            set(user_names_with_keypair).intersection(user_names_tenant)
+
+    def test_tenant_exists_on_dst(self):
+        if TEST_TENANT_NAME in self.tenants:
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: Please get all tenants that must be deleted from config.py

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
+import tests.config as config
+import tests.functional_test as functional_test
+
+TEST_TENANT_NAME = 'tenant3'
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: I think you can remove this constant and use list of deleted tenants from config.py

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        self.dst_volumes_admin = \
+            {vol.display_name: vol for vol in dst_cinder_volumes}
+
+        self.dst_vm_list = \
+            self.dst_cloud.novaclient.servers.list(
+                search_opts={'tenant_id': self.tenants['admin']})
+
+        self.tenant_users_list = \
+            [user for user in config.users if
+             'tenant' in user and user['tenant'] == TEST_TENANT_NAME]
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: change 
... == TEST_TENANT_NAME
to 
... in deleted_tenants_list

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        for volume_name, volume_obj in self.dst_volumes_admin.iteritems():
+            if volume_name in [vol['display_name'] for
+                               vol in src_volumes_list]:
+                if self._volume_is_attached_to_vm(volume_obj):
+                    continue
+                else:
+                    result_vol_ids.append(volume_obj.id)
+        return result_vol_ids
+
+    def test_tenant_cinder_vols_on_dst(self):
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: please rename to test_tenants_volumes_on_dst

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                    if 'key_name' not in src_vm:
+                        continue
+                    if dst_vm.key_name not in migrated_keys:
+                        keys_list.append(dst_vm.key_name)
+        if keys_list:
+            msg = 'Tenant\'s key_pairs {0} exist on destination,' \
+                  ' but should be deleted!'
+            self.fail(msg.format(keys_list))
+
+    def test_tenant_flavs_exist_on_dst(self):
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: flavs to flavours

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.test_verify_dst_del_tenant_resources
devlab/tests/test_verify_dst_del_tenant_resources.py:33: [W0612(unused-variable), VerifyDstDeletedTenantResources.test_tenant_exists_on_dst] Unused variable 'tenant'
devlab/tests/test_verify_dst_del_tenant_resources.py:52: [W0612(unused-variable), VerifyDstDeletedTenantResources.test_tenant_users_exist_on_dst] Unused variable 'tenant'
devlab/tests/test_verify_dst_del_tenant_resources.py:118: [E1101(no-member), VerifyDstDeletedTenantResources.test_tenant_vm_exists_on_dst] Instance of 'list' has no 'attach' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+            self.fail(msg.format(TEST_TENANT_NAME))
+
+    def tenant_users_exist_on_dst(self, users_dict_list):
+        undeleted_users = []
+        dst_users_list = self.dst_cloud.keystoneclient.users.list()
+
+        for src_user_name in [user['name'] for user in users_dict_list]:
+            if src_user_name in dst_users_list:
+                undeleted_users.append(src_user_name)
+        return bool(len(undeleted_users) == len(users_dict_list))
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: remove bool()

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+            [user for user in config.users if
+             'tenant' in user and user['tenant'] == TEST_TENANT_NAME]
+
+        user_names_tenant = [user['name'] for user in self.tenant_users_list]
+        user_names_with_keypair = [key['user'] for key in config.keypairs]
+        self.tenant_users_with_keyp = \
+            set(user_names_with_keypair).intersection(user_names_tenant)
+
+    def tenant_exists_on_dst(self, tenant_name):
+        return bool(tenant_name in self.tenants)
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: same as below

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+    def test_tenant_users_exist_on_dst(self):
+        if self.tenant_users_exist_on_dst(self.tenant_users_list):
+            msg = 'Tenant\'s users {0} exist on destination,' \
+                  ' but should be deleted!'
+            self.fail(msg.format(self.tenant_users_list))
+
+    def net_is_attached_to_vm(self, src_net_name):
+        networks_names_lists = [vm.networks.keys() for vm in self.dst_vm_list]
+        network_names_list = sum(networks_names_lists, [])
+        return bool(src_net_name in network_names_list)
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: remove bool()

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+    def tenant_networks_exist_on_dst(self):
+        dst_nets = self.dst_cloud.neutronclient.list_networks(
+            all_tenants=False)['networks']
+        nets_ids_list = []
+        for dst_net in dst_nets:
+            for src_net in self.tenant_res_dict['networks']:
+                if dst_net['name'] == src_net['name']:
+                    if not self.net_is_attached_to_vm(src_net['name']):
+                        nets_ids_list.append(dst_net['id'])
+        return bool(nets_ids_list), nets_ids_list
```
Comment (plomakin) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: What is the logic to get boolean from list?

**Comment by roman-verchikov** [comment]
Looks ok, but requires SAIC-608 to be resolved first

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+                    match_vm_id += 1
+        return len(vol_obj.attachments) == match_vm_id
+
+    def _tenant_volumes_exist_on_dst(self, src_volumes_list):
+        result_vol_ids = []
+        for volume_name, volume_obj in self.dst_volumes_admin.iteritems():
+            if volume_name in [vol['display_name'] for
+                               vol in src_volumes_list]:
+                if self._volume_is_attached_to_vm(volume_obj):
+                    continue
```
Comment (mirrorcoder) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: Excessive action. Maybe cut so
```
if not self._volume_is_attached_to_vm(volume_obj):
      result_vol_ids.append(volume_obj.id)
```

File: devlab/tests/test_verify_dst_del_tenant_resources.py [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+            for dst_vm in self.dst_vm_list:
+                if dst_vm.id == attached_dev['server_id']:
+                    match_vm_id += 1
+        return len(vol_obj.attachments) == match_vm_id
+
+    def _tenant_volumes_exist_on_dst(self, src_volumes_list):
+        result_vol_ids = []
+        for volume_name, volume_obj in self.dst_volumes_admin.iteritems():
+            if volume_name in [vol['display_name'] for
+                               vol in src_volumes_list]:
```
Comment (mirrorcoder) [inline devlab/tests/test_verify_dst_del_tenant_resources.py]: This code running in loop
```
if volume_name in [vol['display_name'] for vol in src_volumes_list]
```
I think You should rewrite part code in the following way
```
src_vol_list_by_disp_name = [vol['display_name'] for vol in src_volumes_list]
for volume_name, volume_obj in self.dst_volumes_admin.iteritems():
    if volume_name in src_vol_list_by_disp_name:
      .....
```

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by mirrorcoder** [comment]
+1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/config.py [hunk 13]
```diff
@@ -120,6 +122,21 @@
...
+               'routers_to_connect': ['ext_router']}]}],
+     'vms': [
+         {'name': 'tn3server1', 'image': 'image1', 'flavor': 'flavorname1',
+          'key_name': 'key4'}],
+     'cinder_volumes': [
+         {'display_name': 'tn3_volume1', 'size': 1,
+          'server_to_attach': 'tn3server1', 'device': '/dev/vdb'}],
+     'cinder_snapshots': [
+         # Commented because of unimplemented error in nfs driver for grizzly.
+         # {'name': 'tn3snapsh1', 'volume_id': 'tn3_volume1'}
```
Comment (plomakin) [inline devlab/tests/config.py]: please remove it.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by plomakin** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**